src/services/storage.js

```diff
@@ -17,135 +17,135 @@
-Storage.prototype.save = function (id: string, data: Object, forceRewrite: boolean = false) {
-  return this.db().get(id)
-    .catch(err => {
-      if (err.status !== 404) {
-        throw err;
-      }
-      return {};
-    })
-    .then(doc => {
-      if (activeDocs[id]) {
-        Sentry.captureMessage('Race condition spotted', {
-          extra: {
-            id,
-            data,
-            forceRewrite,
-          },
-        });
-      }
+  this.activeDocs[key] = true;
 
-      activeDocs[id] = true;
-      const options = { force: forceRewrite };
-      const record = forceRewrite
-        ? { _id: id, _rev: doc._rev, ...data }
-        : merge(
-          {},
-          doc,
-          {
-            _id: id,
-            _rev: doc._rev,
-          },
-          data,
-        );
-      return this.db().put(record, options);
-    })
-    .then(doc => {
-      activeDocs[id] = false;
-      return doc;
+  return AsyncStorage
+    .setItem(key, JSON.stringify(newValue))
+    .then(() => {
+      this.activeDocs[key] = false;
     })
-    .catch((err) => {
-      if (err.status !== 409) {
-        Sentry.captureException({
-          id,
-          data,
-          err,
-        });
-        throw err;
+    .catch(err => {
+      Sentry.captureException({
+        id,
+        data,
+        err,
+      });
+      if (__DEV__) {
+        console.log(id, err); // eslint-disable-line
       }
-      activeDocs[id] = false;
-      return this.save(id, data, forceRewrite);
+      this.activeDocs[key] = false;
     });
 };
 
-Storage.prototype.getAllDocs = function () {
-  return this.db().allDocs({ conflicts: true });
+Storage.prototype.getAllKeys = function () {
+  return AsyncStorage
+    .getAllKeys()
+    .then(keys => keys.filter(key => key.startsWith(this.prefix)))
+    .catch(() => []);
+};
+
+Storage.prototype.getAll = function () {
+  return this.getAllKeys()
+    .then(keys => AsyncStorage.multiGet(keys)) // [ ['user', 'userValue'], ['key', 'keyValue'] ]
+    .then(values => values.map(([_key, _value]) => ({ [_key]: JSON.parse(_value) }))) // [{ user: 'userValue' }, ...]
+    .catch(() => []);
+};
+
+Storage.prototype.removeAll = async function () {
+  const keys = await this.getAllKeys()
+    .then(data => data.filter(key => key !== STORAGE_SETTINGS_KEY));
+  return AsyncStorage.multiRemove(keys);
 };
 
-Storage.prototype.removeAll = function () {
-  return this.db().allDocs().then(result => {
-    return Promise.all(result.rows.map(row => {
-      return this.db().remove(row.id, row.value.rev);
+Storage.prototype.migrateFromPouchDB = async function () {
+  const { storageSettings = {} } = await this.get(STORAGE_SETTINGS_KEY);
+  if (storageSettings.pouchDBMigrated) return Promise.resolve();
+
+  try {
+    const pouchDBStorage = PouchDBStorage.getInstance('db');
+    const pouchDocs = await pouchDBStorage.getAllDocs()
+      .then(({ rows }) => rows.map(({ doc }) => doc));
+
+    await Promise.all(pouchDocs.map((doc) => {
```

Comment:
If this can be done in batches, even more optimized can be AsyncStorage.multiSet(keyvaluePairs,cb)

Reply:
It's a one-time operation, I think users could wait for an extra second once in their life ;)

Reply:
but the overall suggestion is good ;)